src/ng/compile.js

```diff
@@ -2455,6 +2469,24 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     }
 
 
+    function addEventBindingDirective(directives, eventName, isEventDirective) {
+      if (!isEventDirective) return;
+
+      directives.push({
+        compile: function(tElement, tAttr) {
```

Comment:
oh you're right, didn't notice you were returning a function. Regardless, I was speaking with Igor about this a moment ago and I think we're going to avoid bringing this into angular 1.x, to minimize confusion, and in my view avoid the edge cases which is a big problem